resources/js/Wishlist.vue

```diff
@@ -0,0 +1,199 @@
+<script>
+export default {
+    props: {
+        wishlistId: {
+            type: Number,
+            default: 0
+        },
+        sharedId: String
+    },
+
+    data() {
+        return {
+            wishlists: [],
+            wishlist: null
+        }
+    },
+
+    mounted() {
+        if(this.sharedId) {
+            this.fetchShared();
+        } else {
+            this.fetchWishlists();
+        }
+    },
+
+    methods: {
+        async fetchWishlists() {
+            if (!this.$root.user) {
+                return;
+            }
+
+            if (!localStorage.wishlists) {
+                await this.$root.apiRequest('get', '/api/wishlists/all', {}, function(response) {
+                    localStorage.wishlists = JSON.stringify(response.data);
+                })
+            }
+
+            this.wishlists = JSON.parse(localStorage.wishlists);
+
+            if(this.wishlistId) {
+                this.wishlist = this.wishlists.find(e => e.id == this.wishlistId);
+            }
+        },
+
+        async fetchShared() {
+            this.wishlist = await this.$root.apiRequest('get', '/api/wishlists/shared/' + this.sharedId);
+        },
+
+        getWishlist(id) {
+            var ret = this.wishlists.find(e => e.id == id);
+            if(!ret) return null;
+            if(!ret.title) ret.title = "";
+            if(!ret.description) ret.description = "";
+            if(!ret.share) ret.share = false;
+            return ret;
+        },
+
+        contains(wishlist, productId) {
+            return wishlist.items.find(e => e.product_id == productId);
+        },
+
+        async toggleItem(wishlist, productId, qty = 1, description = '', redirect) {
+            var contains = this.contains(wishlist, productId);
+            if (contains) {
+                this.removeItem(wishlist, contains.wishlist_item_id, redirect);
+            } else {
+                this.addItem(wishlist, productId, qty, description, redirect);
+            }
+        },
+
+        async checkMake(id, title, callback) {
+            var ret = this.getWishlist(id);
```

Comment:
Who is ret? Please use proper variable names and `let` should be enough here: `let wishlist = `